### PR TITLE
[4.0] Remove deprecated code from the Content table class

### PR DESCRIPTION
**Pull request by laoneo**
Remove deprecated code from the Content table class.

### CODE REVIEW

File: libraries/src/Table/Content.php [hunk 1]
```diff
@@ -302,30 +302,6 @@ public function check()
...
 
-	/**
-	 * Gets the default asset values for a component.
-	 *
-	 * @param   string  $component  The component asset name to search for
-	 *
-	 * @return  Rules  The Rules object for the asset
-	 *
-	 * @since   3.4
-	 * @deprecated  3.4 Class will be removed upon completion of transition to UCM
```
Comment (wilsonge) [inline libraries/src/Table/Content.php:316]: We haven't transitioned to UCM so isn't this required?

Comment (laoneo) [reply libraries/src/Table/Content.php:316]: I hoped you can answer that question. If it is still needed, then we should probably remove the deprecated tag as it is very unlikely that this transition will ever happen. On the phone right now, but I think I searched for references as I made the pr but couldn't find one.

Comment (wilsonge) [reply libraries/src/Table/Content.php:316]: I can't find references either at a quick run through. I'll try and do some git blames later this week and try and find out more about it. I'm nervous about things I don't understand :P

Comment (laoneo) [reply libraries/src/Table/Content.php:316]: @wilsonge did you find something?

Comment (wilsonge) [reply libraries/src/Table/Content.php:316]: Yup. We trashed the only use here https://github.com/joomla/joomla-cms/commit/fed63a28cae89db10e0a5d8ab5af5edc3d33897b#diff-86bcd86731ff1e37bf472c0746b4a166L234 (I apparently added this function an even longer time ago :P )

**Comment by wilsonge** [comment]
@SniperSister can you look at rips please :)

**Comment by zero-24** [comment]
Rebooted drone: http://ci.joomla.org/joomla/joomla-cms/12968

**Comment by zero-24** [comment]
It is green now. Thanks

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on e8fcbd2674d1b8819837615386bc360cee3c86d7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22380">issues.joomla.org/tracker/joomla-cms/22380</a>.</sub>

**Comment by wilsonge** [comment]
Thanks! Removing docs required label as it's a protected function.

**Pull request merged**